src/ethereum/prague/fork.py

```diff
@@ -964,7 +968,19 @@ def process_transaction(
 
     output = process_message_call(message, env)
 
+    # For EIP-7623 gas_used includes the intrinsic costs with the floor price
+    # for calldata.
     gas_used = tx.gas - output.gas_left
+
+    # EIP-7623 floor price (note: no EVM costs)
+
+    # Transactions with less gas used than the floor pay at the floor cost.
+    floor = Uint(tokens_in_calldata * FLOOR_CALLDATA_COST + TX_BASE_COST)
+    if gas_used < floor:
+        gas_used = floor
```

Comment:
I believe this correction should be applied *after* the gas refund is applied. Argumentation and spec clarification here: https://github.com/ethereum/EIPs/pull/9227.